lib/web3/formatters.js

```diff
@@ -190,14 +190,20 @@ var outputBlockFormatter = function(block) {
  * @returns {Object} log
 */
 var outputLogFormatter = function(log) {
-    if(log.blockNumber !== null)
-        log.blockNumber = utils.toDecimal(log.blockNumber);
-    if(log.transactionIndex !== null)
-        log.transactionIndex = utils.toDecimal(log.transactionIndex);
-    if(log.logIndex !== null)
-        log.logIndex = utils.toDecimal(log.logIndex);
-
-    return log;
+    if ((log !== null) && (typeof log === 'object')) {
```

Comment:
If I have to be honest log can be also undefined. The problem with typeof is that too many object can have object type. 
Go with an arr for example. typeof []  == "object". I think better way to check if something is object is to use log.toString() ==  "[object Object]"